### PR TITLE
Add consume messages endpoints

### PR DESCRIPTION
**Pull request by EbubeCode**
A client can consume messages, by providing the station as a path variable, the consumer name as a request param and the batch size too. The rest gateway automatically acknowledges the messages received from the Memphis broker.

The Rest gateway caches the message in a database or in memory based on configuration, before responding to the client's request. The client will then send an acknowledgement request by sending an array of message IDs to remove from the cache.

### CODE REVIEW

**Comment by idanasulinmemphis** [comment]
Hi @EbubeCode  thanks for your contribution..
Please add relevant section on the readme so users will be able to know how to use it

**Comment by idanasulinmemphis** [comment]
@EbubeCode please reach me out on email idan@memphis.dev we should talk about it I think the implementation is too complicated and some of the things are not implemented as expected

File: handlers/consumer.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"strconv"
+	"strings"
+	"time"
+)
+
+type requestBody struct {
+	ConsumerName      string `json:"consumer_name"`
+	ConsumerGroupName string `json:"consumer_group_name"`
+	BatchSize         int    `json:"batch_size"`
+	MaxAckTime        int    `json:"max_ack_time"`
```
Comment (idanasulinmemphis) [inline handlers/consumer.go:18]: remove this one

File: handlers/consumer.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	BatchMaxWaitTime  int    `json:"batch_max_wait_time"`
+	MaxMsgDeliveries  int    `json:"max_msg_deliveries"`
```
Comment (idanasulinmemphis) [inline handlers/consumer.go:20]: remove this one

File: handlers/consumer.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	"strings"
+	"time"
+)
+
+type requestBody struct {
+	ConsumerName      string `json:"consumer_name"`
+	ConsumerGroupName string `json:"consumer_group_name"`
+	BatchSize         int    `json:"batch_size"`
+	MaxAckTime        int    `json:"max_ack_time"`
+	BatchMaxWaitTime  int    `json:"batch_max_wait_time"`
```
Comment (idanasulinmemphis) [inline handlers/consumer.go:19]: batch_max_wait_time_ms so people will know it is in ms

File: handlers/consumer.go (hunk 2 above)
Comment (idanasulinmemphis) [reply handlers/consumer.go:20]: and you missed 
startConsumeFromSequence and lastMessages that should also be supported

File: handlers/consumer.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+			log.Errorf("ConsumeHandleMessage - fetch messages: %s", err.Error())
+			c.Status(fiber.StatusInternalServerError)
+			return c.JSON(&fiber.Map{
+				"success": false,
+				"error":   "Server error",
+			})
+		}
+
+		type message struct {
+			Message string `json:"message"`
```
Comment (idanasulinmemphis) [inline handlers/consumer.go:105]: add the message headers

**Comment by idanasulinmemphis** [comment]
Hi @EbubeCode  I left some comments, please fix them so we will be able to merge

File: handlers/consumer.go (hunk 2 above)
Comment (EbubeCode) [reply handlers/consumer.go:20]: I couldn't find the two methods `FetchStartConsumeFromSeq` and `FetchLastMessages` in the SDK
image

**Review by idanasulinmemphis — APPROVED** [review]
@EbubeCode thanks for your neat code, approved and merged

**Pull request merged**